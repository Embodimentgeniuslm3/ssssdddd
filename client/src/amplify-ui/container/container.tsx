import {Component, h, Prop, Host} from "@stencil/core";
import {containerStyle} from "./container.style";

@Component({tag: "amplify-container", shadow: false})
export class AmplifyContainer {
  /*** incase users want to add a class to the inner div */
  @Prop() readonly innerClass?: string;

  render() {
    return (
      <Host class={containerStyle}>
        <div
          class={{
            ...(this.innerClass ? {[this.innerClass]: true} : {}),
          }}
        >
          <slot />
        </div>
      </Host>
    );
  }
}
