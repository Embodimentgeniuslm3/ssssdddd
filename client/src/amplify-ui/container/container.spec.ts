import {AmplifyContainer} from "./container";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-container", () => {
  describe("Component logic", () => {
    let amplifyContainer: AmplifyContainer;
    beforeEach(() => (amplifyContainer = new AmplifyContainer()));

    it("should init `innerClass` as `undefined`", () =>
      expect(amplifyContainer.innerClass).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifyContainer],
            html: `<amplify-container />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
