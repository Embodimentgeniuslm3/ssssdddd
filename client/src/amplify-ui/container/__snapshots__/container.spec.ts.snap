// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-container Render logic should render 1`] = `
<amplify-container class="css-1hbw5z0">
  <!---->
  <div></div>
</amplify-container>
`;
